.github/workflows/docker-build-push-github.yaml

```diff
@@ -0,0 +1,50 @@
+# This job builds a docker image and pushes it to Github's registry.
+# It is intended to be called from other workflows or triggered manually.
+# Based on https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
+
+name: Build Docker image and publish it to the Github registry.
+
+on:
+  # This entry allows calling this workflow from other workflows
+  workflow_call:
+  # This entry allows running this workflow manually
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+ 
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Log in to the Container registry
+        uses: docker/login-action@v2
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          
+      - name: Inspect env vars
+        run: "echo REGISTRY: ${{ env.REGISTRY }}\nIMAGE_NAME: ${{ env.IMAGE_NAME }}"
```

Comment:
```suggestion
        run: |
          echo REGISTRY: ${{ env.REGISTRY }}
          echo IMAGE_NAME: ${{ env.IMAGE_NAME }}
```

Reply:
Not surei f the env vars will be printed this way, but there should be some mechanism to read them anyways.